wpimath/src/main/native/include/units/base.h

```diff
@@ -219,6 +223,28 @@ namespace units
 #else
     #define UNIT_ADD_IO(namespaceName, nameSingular, abbrev)
 #endif
+/**
+ * @def			UNIT_ADD_TELEMETRY(namespaceName,nameSingular, abbreviation)
+ * @brief		Macro for generating the boiler-plate code needed for telemetry for a new unit.
+ * @details		The macro generates the code to insert units into a TelemetryTable
+ * @param		namespaceName namespace in which the new units will be encapsulated.
+ * @param		nameSingular singular version of the unit name, e.g. 'meter'
+ * @param		abbrev - abbreviated unit name, e.g. 'm'
+ * @note		When UNIT_LIB_DISABLE_TELEMETRY is defined, the macro does not generate any code
+ */
+#if __has_include(<wpi/telemetry/TelemetryTable.h>) && !defined(UNIT_LIB_DISABLE_TELEMETRY)
+  #define UNIT_ADD_TELEMETRY(namespaceName, nameSingular, abbrev)\
+  namespace namespaceName\
+  {\
+	  inline void LogTo(wpi::TelemetryTable& table, std::string_view name, const nameSingular ## _t& value)\
+	  {\
+			table.SetProperty(name, "unit", #nameSingular);\
+			table.Log(name, value.value());\
+	  }\
+  }
+#else
+	 #define UNIT_ADD_TELEMETRY(namespaceName, nameSingular, abbrev)
+#endif
```

Comment:
Since `meter_t` and co. are just typedefs of `units::unit_t<meter>`, I think we just need to define a templated `LogTo()` in the `units` namespace after the [`class unit_t` definition](https://github.com/wpilibsuite/allwpilib/blob/e2cc9e0059a011099cdebc3d99e205aa8f4beb3b/wpimath/src/main/native/include/units/base.h#L1928-L2204). We can use `units::name()` (defined by `UNIT_ADD_NAME` below) instead of `#nameSingular` to get the name since we'll no longer be inside the macro.

Reply:
Okay, I’ll give that a try.  I tried minimizing the test case and it worked there, so I’m confused by why this doesn’t work, but avoiding defining it a bunch of times seems better too.

Reply:
Hmm, doesn't work:
```
base.h:116:52: error: inline function 'constexpr const char* units::name(const T&) [with T = unit_t<unit<std::ratio<1>, base_unit<std::ratio<0, 1>, std::ratio<0, 1>, std::ratio<-2>, std::ratio<1> >, std::ratio<0, 1>, std::ratio<0, 1> >, double, linear_scale>]' used but never defined [-Werror]
  116 |         template<typename T> constexpr const char* name(const T&);
```